### PR TITLE
ionice & nice added to mysqldump to fix high cpu resource issue

### PR DESCRIPTION
**Pull request by Anishmourya**
Signed-off-by: Anish Dhanka <anish.mourya5@gmail.com>

### CODE REVIEW

File: CONTRIBUTORS.md [hunk 1]
```diff
@@ -3,3 +3,5 @@
 * Avi Deitcher <https://github.com/deitch>
 * Juan Baptiste <https://github.com/juanluisbaptiste>
 * Luke Barton <https://github.com/lukebarton>
+
+* Anish Dhanka <https://github.com/anishmourya>
```
Comment (deitch) [inline CONTRIBUTORS.md:7]: Please remove the extra line between Luke and you.

File: functions.sh [hunk 2]
```diff
@@ -125,7 +125,11 @@ function do_dump() {
       [ $? -ne 0 ] && return 1
     fi
     for onedb in $DB_NAMES; do
-      mysqldump -h $DB_SERVER -P $DB_PORT $DBUSER $DBPASS --databases ${onedb} $MYSQLDUMP_OPTS > $workdir/${onedb}_${now}.sql
+      if [[ $NICE ]]; then
```
Comment (deitch) [inline functions.sh:128]: Since you documented it as `true`, let's make this:

```sh
if [ "$NICE" = "true" ]; then
```

Comment (deitch) [reply functions.sh:128]: The line is repetitive here, so let's make it cleaner:

```sh
NICE_CMD=
if [ "$NICE" = "true" ]; then
   NICE_CMD="nice -n19 ionice -c2"
fi
$NICE_CMD mysqldump -h ...
```

File: functions.sh [hunk 3]
```diff
@@ -135,7 +139,11 @@ function do_dump() {
     else
       DB_LIST="-A"
     fi
-    mysqldump -h $DB_SERVER -P $DB_PORT $DBUSER $DBPASS $DB_LIST $MYSQLDUMP_OPTS > $workdir/backup_${now}.sql
+    if [[ $NICE ]]; then
+        nice -n19 ionice -c2 mysqldump -h $DB_SERVER -P $DB_PORT $DBUSER $DBPASS $DB_LIST $MYSQLDUMP_OPTS > $workdir/backup_${now}.sql
+    else
+       mysqldump -h $DB_SERVER -P $DB_PORT $DBUSER $DBPASS $DB_LIST $MYSQLDUMP_OPTS > $workdir/backup_${now}.sql
+    fi
```
Comment (deitch) [inline functions.sh:146]: Same comments as for the above mysqldump command.

**Review by deitch — CHANGES_REQUESTED** [review]
Thanks for this, looks good. A few changes, and we will be good to go.

**Comment by Anishmourya** [comment]
@deitch thanks for pr review. I have updated according to your comments. let me know if I need to change something else. thanks

**Review by deitch — APPROVED** [review]
Thanks!

**Pull request merged**